application/src/main/java/org/togetherjava/tjbot/commands/moderation/NoteCommand.java

```diff
@@ -0,0 +1,106 @@
+
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * This command allows users to write notes about others. Notes are persisted and can be retrieved
+ * using {@link AuditCommand}, like other moderative actions.
+ * <p>
+ * The command fails if the user triggering it is lacking permissions to either write a note about
+ * other users or to write a note about the specific given user (for example a moderator attempting
+ * to write a note about an admin).
+ */
+public final class NoteCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(NoteCommand.class);
+    private static final String USER_OPTION = "user";
+    private static final String CONTENT_OPTION = "content";
+    private static final String ACTION_VERB = "write a note about";
+    private final ModerationActionsStore actionsStore;
+    private final Predicate<String> hasRequiredRole;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param actionsStore used to store actions issued by this command
+     * @param config the config to use for this
+     */
+    public NoteCommand(@NotNull ModerationActionsStore actionsStore, @NotNull Config config) {
+        super("note", "Writes a note about the given user", SlashCommandVisibility.GUILD);
+
+        getData()
+            .addOption(OptionType.USER, USER_OPTION, "The user who you want to write a note about",
+                    true)
+            .addOption(OptionType.STRING, CONTENT_OPTION,
+                    "The content of the note you want to write", true);
+
+        hasRequiredRole = Pattern.compile(config.getSoftModerationRolePattern()).asMatchPredicate();
+        this.actionsStore = Objects.requireNonNull(actionsStore);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        OptionMapping targetOption = event.getOption(USER_OPTION);
+        Member author = event.getMember();
+        Guild guild = event.getGuild();
+        String content = event.getOption(CONTENT_OPTION).getAsString();
+
+        if (!handleChecks(guild.getSelfMember(), author, targetOption.getAsMember(), content,
+                event)) {
+            return;
+        }
+
+        sendNote(targetOption.getAsUser(), author, content, guild, event);
+    }
+
+    @SuppressWarnings("BooleanMethodNameMustStartWithQuestion")
+    private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
+            @Nullable Member target, CharSequence content, @NotNull Interaction event) {
+        if (target != null && !ModerationUtils.handleCanInteractWithTarget(ACTION_VERB, bot, author,
+                target, event)) {
+            return false;
+        }
+        if (!ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event)) {
+            return false;
+        }
+        return ModerationUtils.handleReason(content, event);
```

Comment:
```suggestion
        if (target != null && !ModerationUtils.handleCanInteractWithTarget(ACTION_VERB, bot, author,
                target, event)) {
            return false;
        }
        
        if (!ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event)) {
            return false;
        }
        
        return ModerationUtils.handleReason(content, event);
```

Yes it's me again, like always with the styling :p

Reply:
fair